astronomer/providers/amazon/aws/operators/emr.py

```diff
@@ -1,95 +1,22 @@
-        job_id = hook.submit_job(
-            name=self.name,
-            execution_role_arn=self.execution_role_arn,
-            release_label=self.release_label,
-            job_driver=self.job_driver,
-            configuration_overrides=self.configuration_overrides,
-            client_request_token=self.client_request_token,
-            tags=self.tags,
-        )
-        try:
-            # for apache-airflow-providers-amazon<6.0.0
-            polling_attempts = self.max_tries  # type: ignore[attr-defined]
-        except AttributeError:  # pragma: no cover
-            # for apache-airflow-providers-amazon>=6.0.0
-            # max_tries is deprecated so instead of max_tries using self.max_polling_attempts
-            polling_attempts = self.max_polling_attempts
-
-        query_state = hook.check_query_status(job_id)
-        if query_state in hook.SUCCESS_STATES:
-            self.log.info(
-                f"Try : Query execution completed. Final state is {query_state}"
-                f"EMR Containers Operator success {query_state}"
-            )
-            return job_id
-
-        if query_state in hook.FAILURE_STATES:
-            error_message = self.hook.get_job_failure_reason(job_id)
-            raise AirflowException(
-                f"EMR Containers job failed. Final state is {query_state}. "
-                f"query_execution_id is {job_id}. Error: {error_message}"
-            )
-
-        self.defer(
-            timeout=self.execution_timeout,
-            trigger=EmrContainerOperatorTrigger(
-                virtual_cluster_id=self.virtual_cluster_id,
-                job_id=job_id,
-                aws_conn_id=self.aws_conn_id,
-                poll_interval=self.poll_interval,
-                max_tries=polling_attempts,
+    def __init__(self, *args, **kwargs):  # type: ignore[no-untyped-def]
+        warnings.warn(
+            (
+                "This module is deprecated. "
+                "Please use `airflow.providers.amazon.aws.operators.emr.EmrContainerOperator` "
+                "and set deferrable to True instead."
             ),
-            method_name="execute_complete",
+            DeprecationWarning,
+            stacklevel=2,
         )
-
-        # for bypassing mypy missing return error
-        return None  # pragma: no cover
-
-    def execute_complete(self, context: Context, event: dict[str, Any]) -> str:  # type: ignore[override]
-        """
-        Callback for when the trigger fires - returns immediately.
-        Relies on trigger to throw an exception, otherwise it assumes execution was
-        successful.
-        """
-        if "status" in event and event["status"] == "error":
-            raise AirflowException(event["message"])
-        self.log.info(event["message"])
-        job_id: str = event["job_id"]
-        return job_id
+        return super().__init__(*args, deferrable=True, **kwargs)
```

Comment:
Do we need a return here?

Reply:
Nope, i think it's accidentally added. let me send a quick fix

Reply:
https://github.com/astronomer/astronomer-providers/pull/1430